orc8r/cloud/go/obsidian/server/metrics.go

```diff
@@ -50,12 +50,21 @@ func CollectStats(next echo.HandlerFunc) echo.HandlerFunc {
 		requestCount.Inc()
 		status := strconv.Itoa(c.Response().Status)
 		respStatuses.WithLabelValues(status, c.Request().Method).Inc()
+		url := sanitizeString(c.Request().URL.String())
 		glog.V(2).Infof(
 			"REST API code: %v, method: %v, url: %v\n",
 			status,
 			c.Request().Method,
-			c.Request().URL,
+			url,
 		)
 		return nil
 	}
 }
+
+func sanitizeString(strString string) string {
+	strSanitizedString := html.EscapeString(strString) //escape special charatchters in HTML text
+	strSanitizedString = strings.Replace(strSanitizedString, "\r", "", -1) //remove line breaks
+	strSanitizedString = strings.Replace(strSanitizedString, "\n", "", -1)
+	strSanitizedString = strings.Join(strings.Fields(strSanitizedString), " ") //remove extra whitespace
+	return strSanitizedString
+}
```

Comment:
when did you see an url containing `\r` or `\n`?  If that is the case maybe there are another issue here

Reply:
I threw out  strings.Replace for \r and \n...